frame/staking/src/tests.rs

```diff
@@ -3084,19 +3088,16 @@ fn expired_ring_should_capable_to_promise_again() {
 		// REVIEW: NotStlash problem
 		// TODO: check on this
 		// left: `Err(DispatchError::Module { index: 0, error: 1, message: Some("NotStash") })`,
-		// assert_ok!(Staking::deposit_extra(
-		// 	Origin::signed(controller),
-		// 	promise_extra_value,
-		// 	13,
-		// ));
-		// ledger.active_deposit_ring = promise_extra_value;
-		// // old deposit_item with 12 months promised removed
-		// ledger.deposit_items = vec![TimeDepositItem {
-		// 	value: promise_extra_value,
-		// 	start_time: ts,
-		// 	expire_time: 2 * ts,
-		// }];
-		// assert_eq!(Staking::ledger(controller).unwrap(), ledger);
+		// @clearloop: Fixed.
+		assert_ok!(Staking::deposit_extra(Origin::signed(stash), promise_extra_value, 13,));
+		ledger.active_deposit_ring = promise_extra_value;
+		// old deposit_item with 12 months promised removed
+		ledger.deposit_items = vec![TimeDepositItem {
+			value: promise_extra_value,
+			start_time: ts,
+			expire_time: 2 * ts,
+		}];
+		assert_eq!(Staking::ledger(controller).unwrap(), ledger);
```

Comment:
Most of `NotStash` problem TODOs caused by the dislocations between `stash` and `controller`